pandas/core/frame.py

```diff
@@ -2727,11 +2727,16 @@ def set_index(self, keys, drop=True, append=False, inplace=False,
                   verify_integrity=False):
         """
         Set the DataFrame index (row labels) using one or more existing
-        columns. By default yields a new object.
+        columns and/or new arrays of values. By default yields a new object.
 
         Parameters
         ----------
-        keys : column label or list of column labels / arrays
+        keys : column label (str), Index, Series, array, or a list of these things
+            Existing columns to set as the index (when given columns labels)
+            and/or new values to set as the index. If an Index is given, it's
```

Comment:
@jreback - So, for any list-like, if all the elements are valid column labels, add these columns to the index. If not, try to create an index from the values in the list-like. Did I understand that right?